### ISSUE TITLE
With autostart = false, server does not attach to new buffers when project path contains whitespace

### ISSUE THREAD

**Issue by Diomendius** [neovim/nvim-lspconfig#1170]
- `nvim --version`: 0.5.0
- nvim-lspconfig version(commit hash): 5b0fa84ee35006e06142f98b8a5b28d79cfa5000
- What language server (If the problem is related to a specific language server): rust-analyzer, but likely server-agnostic.
- Can you reproduce this behavior on other language server clients (vscode, languageclient-neovim, coc.nvim, etc.): coc.nvim at least does not exhibit this behaviour.
- Can you reproduce this behavior on other language servers offered in the nvim-lspconfig repo? (pyls -> pyright): Untested, but likely server-agnostic.
- Is the problem isolated to a particular language server: Not as far as I can reason.
- Operating system/version: Arch Linux, up to date as of 2021-08-09

### How to reproduce the problem from neovim startup
- Set `autostart = false` in your nvim config for the language server you will use.
- `fallocate -l 128M img && mkntfs -F img && udisksctl loop-setup -f img && udisksctl mount -b /dev/loop#` will create an NTFS image and map a loop device to it, assuming udisks and ntfs-3g are installed. Mount this image with `udisksctl mount -b <device>` where `<device>` is the loop device specified in the stdout of `udisksctl`. Use `sudo losetup` and `sudo mount` instead of `udisksctl` if udisks is not installed.
- Open a project the language server is compatible with, after creating it on or copying it to the mounted NTFS filesystem.
- Run `:LspStart`.
- Check that the language server is functioning and any keybindings triggered within an `on_attach` hook are enabled.
- Run `:e` on another file in the project. (`:e` on its own also triggers this issue in the same way, except that `on_attach` keybindings are not cleared)
- Observe that language server functions now do nothing, `on_attach` keybindings are now unbound and `:LspInfo` shows 0 servers attached to the current buffer.
- `udisksctl unmount -b /dev/loop# && udisksctl loop-delete -b /dev/loop#`

### Actual behaviour
When a language server is configured with `autostart = false` and a project on an NTFS-3G mount is opened, the server becomes detached from the current buffer when a file is loaded into it. New windows opened on other files under the same project root are not attached to the server. This may apply to other FUSE-based filesystems, but I have not tested any others.

### Expected behaviour
If a language server is already running, the server should be automatically attached to any buffer opened on a file under the same project root, regardless of the `autostart` setting.

### Minimal init.vim or init.lua and code sample
```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]

local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'

local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      'neovim/nvim-lspconfig',
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
    },
  }
end

_G.load_config = function()
  vim.lsp.set_log_level 'trace'
  local nvim_lsp = require 'lspconfig'
  local on_attach = function(_, bufnr)
    local function buf_set_keymap(...)
      vim.api.nvim_buf_set_keymap(bufnr, ...)
    end
    local function buf_set_option(...)
      vim.api.nvim_buf_set_option(bufnr, ...)
    end

    buf_set_option('omnifunc', 'v:lua.vim.lsp.omnifunc')

    -- Mappings.
    local opts = { noremap = true, silent = true }
    buf_set_keymap('n', 'gD', '<Cmd>lua vim.lsp.buf.declaration()<CR>', opts)
    buf_set_keymap('n', 'gd', '<Cmd>lua vim.lsp.buf.definition()<CR>', opts)
    buf_set_keymap('n', 'K', '<Cmd>lua vim.lsp.buf.hover()<CR>', opts)
    buf_set_keymap('n', 'gi', '<cmd>lua vim.lsp.buf.implementation()<CR>', opts)
    buf_set_keymap('n', '<C-k>', '<cmd>lua vim.lsp.buf.signature_help()<CR>', opts)
    buf_set_keymap('n', '<space>wa', '<cmd>lua vim.lsp.buf.add_workspace_folder()<CR>', opts)
    buf_set_keymap('n', '<space>wr', '<cmd>lua vim.lsp.buf.remove_workspace_folder()<CR>', opts)
    buf_set_keymap('n', '<space>wl', '<cmd>lua print(vim.inspect(vim.lsp.buf.list_workspace_folders()))<CR>', opts)
    buf_set_keymap('n', '<space>D', '<cmd>lua vim.lsp.buf.type_definition()<CR>', opts)
    buf_set_keymap('n', '<space>rn', '<cmd>lua vim.lsp.buf.rename()<CR>', opts)
    buf_set_keymap('n', 'gr', '<cmd>lua vim.lsp.buf.references()<CR>', opts)
    buf_set_keymap('n', '<space>e', '<cmd>lua vim.lsp.diagnostic.show_line_diagnostics()<CR>', opts)
    buf_set_keymap('n', '[d', '<cmd>lua vim.lsp.diagnostic.goto_prev()<CR>', opts)
    buf_set_keymap('n', ']d', '<cmd>lua vim.lsp.diagnostic.goto_next()<CR>', opts)
    buf_set_keymap('n', '<space>q', '<cmd>lua vim.lsp.diagnostic.set_loclist()<CR>', opts)
  end

  -- Add the server that troubles you here
  local name = 'rust_analyzer'
  local cmd = { 'rust-analyzer' } -- needed for elixirls, omnisharp, sumneko_lua

  nvim_lsp[name].setup {
    cmd = cmd,
    autostart = false,
    on_attach = on_attach,
  }

  print [[You can find your log at $HOME/.cache/nvim/lsp.log. Please paste in a github issue under a details tag as described in the issue template.]]
end

if vim.fn.isdirectory(install_path) == 0 then
  vim.fn.system { 'git', 'clone', 'https://github.com/wbthomason/packer.nvim', install_path }
  load_plugins()
  require('packer').sync()
  vim.cmd [[autocmd User PackerComplete ++once lua load_config()]]
else
  load_plugins()
  require('packer').sync()
  _G.load_config()
end
```

### Health check
```
health#lspconfig#check
========================================================================
## Checking language server protocol configuration
  - INFO: rust_analyzer: configuration checked.
```

### LSP log
[lsp.log](https://github.com/neovim/nvim-lspconfig/files/7033714/lsp2.log)

**Comment by mjlbach** [comment]
I'm not going to fix this issue as I have no way to reproduce. PRs welcome.

**Comment by Diomendius** [comment]
On revisiting this, I think I have a better idea what causes this bug. I can no longer reproduce this myself on the projects I previously could, but since I opened this issue I have changed my filesystem to remove whitespace from the path to my projects directory, as this caused issues with building Nvim. Further testing confirms the bug is not present when the path is for instance `~/Desktop/foo-bar/project` but is present when the path is `~/Desktop/foo bar/project`.

I don't know how I missed this when trying to narrow down the cause originally, as I did try to rule out other causes such as accessing the project behind a symlink, but evidently I missed the obvious.

Hopefully this is enough to reproduce the issue.
